Panic on compaction

I'm seeing a series of panics on our logs related to compaction. Here's the full trace:

```
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x120 pc=0x829c49]
goroutine 495 [running]:
github.com/bsm/myapp/vendor/github.com/prometheus/tsdb.(*LeveledCompactor).write(0xc42ef7caf0, 0xc42019a2b0, 0xe, 0xc49040aa10, 0xc4dff0fcd0, 0x1, 0x1, 0xc6d040, 0xc4852e2bd0)
        /go/src/github.com/bsm/myapp/vendor/github.com/prometheus/tsdb/compact.go:431 +0x579
github.com/bsm/myapp/vendor/github.com/prometheus/tsdb.(*LeveledCompactor).Write(0xc42ef7caf0, 0xc42019a2b0, 0xe, 0xc72380, 0xc4729b3ca0, 0x16122e3a600, 0x16123518300, 0x0, 0x0, 0xccf248, ...)
        /go/src/github.com/bsm/myapp/vendor/github.com/prometheus/tsdb/compact.go:362 +0x2ad
github.com/bsm/myapp/vendor/github.com/prometheus/tsdb.(*DB).compact(0xc4205ff980, 0xc43550ca00, 0x0, 0x0)
        /go/src/github.com/bsm/myapp/vendor/github.com/prometheus/tsdb/db.go:360 +0x21d
github.com/bsm/myapp/vendor/github.com/prometheus/tsdb.(*DB).run(0xc4205ff980)
        /go/src/github.com/bsm/myapp/vendor/github.com/prometheus/tsdb/db.go:263 +0x30a
created by github.com/bsm/myapp/vendor/github.com/prometheus/tsdb.Open
        /go/src/github.com/bsm/myapp/vendor/github.com/prometheus/tsdb/db.go:227 +0x589
```

It looks like it's happening in the line below when `indexw` is `nil` and `index.Version` panics:

```
	indexw, err := index.NewWriter(filepath.Join(tmp, indexFilename))
	meta.Version = indexw.Version
	if err != nil {
		return errors.Wrap(err, "open index writer")
	}
```

--- a/compact.go
+++ b/compact.go
@@ -428,10 +428,10 @@ func (c *LeveledCompactor) write(dest string, meta *BlockMeta, blocks ...BlockRe
 	}
 
 	indexw, err := index.NewWriter(filepath.Join(tmp, indexFilename))
-	meta.Version = indexw.Version
 	if err != nil {
 		return errors.Wrap(err, "open index writer")
 	}
+	meta.Version = indexw.Version
 
 	if err := c.populateBlock(blocks, meta, indexw, chunkw); err != nil {
 		return errors.Wrap(err, "write compaction")

